value.go

```diff
@@ -285,7 +285,16 @@ func (i valueInt) ToNumber() Value {
 }
 
 func (i valueInt) SameAs(other Value) bool {
```

Comment:
[q] Not sure if you had the chance to kind of dig into this but I see there are 3 different ways to compare values:
* SameAs
* Equals
* StrictEquals

`Equals` vs `StrictEquals` I think is the same as `==` vs `===`. So now I'm wondering what `SameAs` is for and maybe we should have just used the `StrictEquals` method inside for the includes logic ?

Reply:
Yeah so a few objects just call SameAs in their Equals method like String and Symbol.

Other than that I just see SameAs used the array includes method and ordered map lookup. I think some of the variability we see between goja and normal js that we called out above could be fixed by modifying some of the Equals methods but might be better served in a separate/future ticket (and might actually be better used as a contribution to the open source repo).

At least in goja it seems like SameAs is less strict than Equals and StrictEquals in this scenario so I am fine limiting the changes to here for now

Reply:
Alrighty I think we can refactor the `SameAs` like so then:

```
func (i valueInt) SameAs(other Value) bool {
	return i.StrictEquals(other)
}
```

Reply:
Agreed, thanks for calling that out